File: Sort Array by Increasing Frequency.py
class Solution:
    def frequencySort(self, nums: List[int]) -> List[int]:
        counter = Counter(nums)
        arr = []
        for k, v in counter.items():
            arr.append((v, -k))
        arr.sort()
        ans = []
        for (v, k) in arr:
            ans += [-k] * v
        return ans

File: Sort Array by Increasing Frequency.cpp
class Solution {
public:
    vector<int> frequencySort(vector<int>& nums) {
        map<int, int> counter;
        for (int num: nums) {
            counter[num]++;
        }

        struct compare
        {
            map<int, int>* counter_;
            compare(map<int, int>* counter) : counter_(counter) {}

            inline bool operator() (int i, int j)
            {
                if ((*counter_)[i] == (*counter_)[j]) return i > j;
                return (*counter_)[i] < (*counter_)[j];
            }
        };

        sort(nums.begin(), nums.end(), compare(&counter));
        return nums;
    }
};
